marklogic-rdf4j/src/main/java/com/marklogic/semantics/rdf4j/MarkLogicRepositoryConnection.java

```diff
@@ -676,7 +677,7 @@ public boolean hasStatement(Resource subject, IRI predicate, Value object, boole
             logger.debug(queryString);
             MarkLogicBooleanQuery query = prepareBooleanQuery(queryString); // baseuri ?
 
-            setBindings(query, subject, predicate, object, contexts);
+            setBindings(query, subject, predicate, Util.getInstance().skolemize(object), contexts);
```

Comment:
After pondering this issue, I realize that, for query, you cannot use skolemized bnodes.
Rather, for each bnode in a given set of statements, you need a free variable with no binding.

I'd like to talk this over with you to get on the same page.  The reason why you cannot use skolemization here is that the same bnode will never skolemize to the same value twice.  This will never match a statement